src/StreamFetcher.ts

```diff
@@ -41,81 +50,87 @@ async function handleNon2xxResponse(funcName: Todo, response: Todo, streamId: st
 export class StreamFetcher {
 
     apiUrl: string
-    fetch: Todo
-    checkPermission: Todo
-    authenticate: Todo
```

Comment:
Removing the `Todo` here actually gives you the correct type automatically via inference.

Reply:
Agreed, though sometimes it's a pain in the ass to figure out where to import the types from if they reference other types. This is basically just a one step up from `Todo` such that autocomplete and typechecking actually works.